### ISSUE TITLE
`doctl copute volume ls -o json` returns null.

### ISSUE THREAD

**Issue by fleeto** [digitalocean/doctl#479]
~~~command
$ doctl compute volume ls -o json
null

$ doctl compute load-balancer ls -o json
[]
~~~

**Comment by hilary** [comment]
@bentranter Feels like we should be able to prevent this sort of error by treating any erroneous nil slices as empty slices in the JSON displaying logic. What do you think?

**Comment by bentranter** [comment]
@hilary I'm not too sure. In our JSON displaying logic, we use `json.Marshal`, and the only option we get there is to either write `null` or nothing in the output, as there's no option for displaying `[]` instead of `null`.

With that said, another approach would be to check each `item` in the `writeJSON` function for any nil slices, and change them to empty slices before calling `json.Marshal`. That would require some pretty heavy use of the `reflect` package though, which I don't think we should get into.

Unfortunately, I think our best bet is to go through each of the commands, and check for cases where we're accidentally returning nil slices. It's the simplest solution, and after briefly looking last week when this issue was reported, I don't think there are too many cases where we do return nil slices.

On a different note, I wonder if fixing this issue qualifies as a breaking change. For example, if there are users who use a tool like `jq` to parse our JSON output, they're probably expecting `null` where we return it, and not `[]`.

**Comment by fleeto** [comment]
But why do I saw `[]` in the load-balancer list?

**Comment by bentranter** [comment]
@fleeto You were seeing `[]` in the load-balancer list due to an inconsistency in the way we rendered empty and nil slices. This is fixed as of [the latest release (1.22.0)](https://github.com/digitalocean/doctl/releases).
